### PR TITLE
Use general OSError and more Python3 defaults from pyupgrade

### PR DESCRIPTION
**Pull request by eumiro**
## PR Summary

Running `pyupgrade --py3-plus` and ignoring all string updates (percent/format/f) to use defaults introduced in Python3.

## PR Checklist

- [N/A] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A] New features are documented, with examples if plot related.
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: versioneer.py [hunk 1]
```diff
@@ -339,7 +339,6 @@
 
 """
 
-from __future__ import print_function
```
Comment (timhoffm) [inline versioneer.py:342]: Please don't change anything in `versioneer.py`. It's a vendored file from (https://github.com/python-versioneer/python-versioneer) with slight modifications. Additional changes make it harder to pull in changes from the upstream version.

**Review by timhoffm — COMMENTED** [review]
Generally :+1: apart from the changes to `versioneer.py`.

File: versioneer.py (hunk 1 above)
Comment (eumiro) [reply versioneer.py]: Thank you @timhoffm . I have removed `versioneer.py` from my PR entirely.

File: lib/matplotlib/_version.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def run_command(commands, args, cwd=None, verbose=False, hide_stderr=False):
                                  stderr=(subprocess.PIPE if hide_stderr
                                          else None))
             break
-        except EnvironmentError:
+        except OSError:
```
Comment (QuLogic) [inline lib/matplotlib/_version.py:74]: I would not bother changing this file either as it will be deleted by #18971 .

**Review by QuLogic — COMMENTED** [review]
All the `raise OSError('` for missing files should probably be `FileNotFoundError` instead.

**Pull request closed without merging**